index.html

```diff
@@ -1565,83 +1613,122 @@
                     Otherwise reject <var>promise</var> with
                     <code>"SyntaxError"</code>, and abort these steps.
                   </li>
-                </ul>
+                </ol>
               </li>
               <li>
                 Add <var>ndef</var> to <var>output</var>.
               </li>
-            </ul>
+            </ol>
           </li> <!-- guessing record.type from the data -->
           <li>
-            Let <var>web ndef</var> be the result of invoking
+            Let <var>webnfc</var> be the result of invoking
             <a>create a Web NFC record</a>.
           </li>
           <li>
-            Add <var>web ndef</var> to <var>output</var>.
+            Add <var>webnfc</var> to <var>output</var>.
+            <p class="note">
+              Implementations may choose the location of the Web NFC record
+              within the <a>NDEF message</a>.
+            </p>
           </li>
           <li>
             If <var>options</var>.timeout value is not valid or not supported
             in the underlying platform, reject <var>promise</var> with
             <code>"SyntaxError"</code>, and abort these steps.
+          </li>
           <li>
-            If <var>options</var>.timeout value is not <code>Infinity</code>,
-            let <var>timer</var> be a new timer associated with this invocation
-            of <code>pushMessage()</code>, set to <var>options</var>.timeout,
-            and start <var>timer</var>. On any termination of this algorithm,
-            <var>timer</var> MUST be cleared.
+            If <var>target</var> has the value <code>"tag"</code> or
+            <code>"any"</code>, then set <var>this@[[\tagMessage]]</var>
+            to <var>output</var>.
+          </li>
+          <li>
+            If <var>target</var> has the value <code>"peer"</code> or
+            <code>"any"</code>, then set <var>this@[[\peerMessage]]</var>
+            to <var>output</var>.
           </li>
           <li>
-            If <var>timer</var> exists and expires, reject <var>promise</var>
-            with <code>"TimeoutError"</code> and abort these steps.
+            If <var>options</var>.timeout value is not <code>Infinity</code>,
+            start the timer <var>this@[[\timer]]</var> with the timeout value
```

Comment:
Since 2 copies of this algorithm could be running at any given time (for "peer" and "tag" targets), you may need 2 timers. Or you could just treat the timer as a local variable here, and fix up references to it from elsewhere in the spec to talk about active pushMessage() executions.